x/upgrade/types/keys.go

```diff
@@ -25,6 +25,9 @@ const (
 	// VersionMapByte is a prefix to look up module names (key) and versions (value)
 	VersionMapByte = 0x2
 
+	// ProtocolVersionByte is a prefix to look up Protocol Version
+	ProtocolVersionByte = 0x3
```

Comment:
We might want to add an InitGenesis and ExportGenesis for this

Reply:
I can add these in. As for InitGenesis I'll have to implement some exported function in the keeper to set the version to state. Thoughts on implementing some `InitializeAndSeal` style function on the keeper so that it can only be called once? I feel like that might be the safest way to go about it, however it would add an additional field on the keeper `sealed`.

Reply:
I'm still thinking if it's really necessary to add the protocol version inside the GenesisState. Maybe hardcoding it inside an upgrade handler is enough.

I propose to leave this PR as-is for now. I'll create an issue tomorrow about more general improvements for the upgrade migrations, i'll mention this in that issue.